toolchains/msvc.toolchain.cmake

```diff
@@ -26,20 +28,25 @@ set( TNUN_compiler_exceptions_off        -wd4577
 set( TNUN_compiler_report_optimization   -Qpar-report:1 -Qvec-report:2               )
 set( TNUN_compiler_optimize_for_speed    -Ox -Ot -Qpar                               ) # https://msdn.microsoft.com/en-us/library/jj658585.aspx Vectorizer and Parallelizer Messages
 set( TNUN_compiler_runtime_sanity_checks -GS -sdl -guard:cf -fp:strict -RTC1 -RTCc -D_ALLOW_RTCc_IN_STL ) # https://www.reddit.com/r/cpp/comments/46mhne/rtcc_rejects_conformant_code_with_visual_c_2015
+set( TNUN_disabled_warnings              -wd4324 -wd4373 -wd4100 -wd4456 -wd4067 -wd4995 -wd4635 -wd4800 -wd4101 -wd4756 -wd4068 -wd4056 -wd4996 -wd4503 )
+set( TNUN_warnings_as_errors             -WX )
+set( TNUN_default_warnings               -W4 )
 
-set( TNUN_compiler_release_flags -DNDEBUG -Bt -Ox -Ob2 -Oy -GF -Gw -Gm- -GS- -Gy )
 
 # w4373: '...': virtual function overrides '...', previous versions of the compiler did not override when parameters only differed by const/volatile qualifiers
 # w4324: 'structure was padded due to alignment specifier'
-add_compile_options( /std:c++latest -MP -Oi -W4 -Zc:threadSafeInit- -wd4324 -wd4373 -wd4100 -wd4456 -wd4067 -wd4995 -wd4635 -wd4800 -wd4101 -wd4756 -wd4068 -wd4056 -wd4996 -wd4503)
+add_compile_options( /std:c++latest -MP -Oi -Zc:threadSafeInit- )
```

Comment:
We treat all warnings as errors and W4 introduces some very silly warnings which would then either need to be disabled with TNUN_disabled_warnings or fixed (in our source this would mean several weeks of work).

TNUN_disabled_warnings variable is added to contain silly warnings that should be disabled because they are both unimportant and difficult or impossible to fix. We treat other warnings as errors so adding those would make code uncompilable. AFAIK, MSVC does not have the ability to treat all warnings as errors, except some specifically defined (something like -Wno-error= on GCC/Clang). Even in that case, producing a lot of warnings, especially in third party libs, would produce a lot of noise in compile output, making it difficult for developers to find actual compile error they are facing (not everyone is using VS which separates warnings from errors in GUI).